File: source/aoBlogs2/Controllers/BlogBodyController.vb

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Text.RegularExpressions
Imports Contensive.Addons.Blog.Controllers
Imports Contensive.Addons.Blog.Models
Imports Contensive.BaseClasses

Namespace Views
    '
    Public Class BlogBodyController
        '
        '====================================================================================
        '
        Public Shared Function ProcessForm(cp As CPBaseClass, app As ApplicationEnvironmentModel, request As View.RequestModel, ByRef RetryCommentPost As Boolean) As Integer
            '
            Try
                If request.ButtonValue <> "" Then
                    Select Case request.SourceFormID
                        Case FormBlogPostList
                            ProcessForm = FormBlogPostList
                        Case FormBlogEntryEditor
                            ProcessForm = EditView.ProcessFormBlogEdit(cp, app, request)
                        Case FormBlogPostDetails
                            ProcessForm = ArticleView.processArticleView(cp, app, request, RetryCommentPost)
                        Case FormBlogArchiveDateList
                            ProcessForm = FormBlogArchiveDateList
                        Case FormBlogSearch
                            ProcessForm = FormBlogSearch
                        Case FormBlogArchivedBlogs
                            ProcessForm = FormBlogArchivedBlogs
                    End Select
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
        End Function
    End Class
End Namespace

File: source/aoBlogs2/Controllers/BlogBodyController.cs

using System;
using Contensive.Blog.Models;
using Contensive.BaseClasses;
using Contensive.Blog.Views;

namespace Contensive.Blog.Controllers {
    // 
    public class BlogBodyController {
        // 
        // ====================================================================================
        // 
        public static int ProcessForm(CPBaseClass cp, ApplicationEnvironmentModel app, Models.View.RequestModel request, ref bool RetryCommentPost) {
            int ProcessFormRet = default;
            // 
            try {
                if (!string.IsNullOrEmpty(request.ButtonValue)) {
                    switch (request.SourceFormID) {
                        case constants.FormBlogPostList: {
                                ProcessFormRet = constants.FormBlogPostList;
                                break;
                            }
                        case constants.FormBlogEntryEditor: {
                                ProcessFormRet = EditView.ProcessFormBlogEdit(cp, app, request);
                                break;
                            }
                        case constants.FormBlogPostDetails: {
                                ProcessFormRet = ArticleView.processArticleView(cp, app, request, ref RetryCommentPost);
                                break;
                            }
                        case constants.FormBlogArchiveDateList: {
                                ProcessFormRet = constants.FormBlogArchiveDateList;
                                break;
                            }
                        case constants.FormBlogSearch: {
                                ProcessFormRet = constants.FormBlogSearch;
                                break;
                            }
                        case constants.FormBlogArchivedBlogs: {
                                ProcessFormRet = constants.FormBlogArchivedBlogs;
                                break;
                            }
                    }
                }
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }

            return ProcessFormRet;
        }
    }
}